homeassistant/components/gree/bridge.py

```diff
@@ -0,0 +1,40 @@
+"""Helper and wrapper classes for Gree module."""
+import logging
+from typing import List
+
+from greeclimate.device import Device
+from greeclimate.device_info import DeviceInfo
+from greeclimate.exceptions import DeviceNotBoundError
+from greeclimate.gree_climate import GreeClimate
+
+from homeassistant import exceptions
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DeviceHelper:
+    """Device search and bind wrapper for Gree platform."""
+
+    @staticmethod
+    async def try_bind_device(device_info: DeviceInfo) -> Device:
+        """Try and bing with a discovered device.
+
+        Note the you must bind with the device very quickly after it is discovered, or the
+        process may not be completed correctly, raising a `CannotConnect` error.
+        """
+        try:
+            device = Device(device_info)
+            await device.bind()
+            return device
+        except DeviceNotBoundError as exception:
+            raise CannotConnect from exception
+
+    @staticmethod
+    async def find_devices() -> List[DeviceInfo]:
+        """Gather a list of device infos from the local network."""
+        gree = GreeClimate()
+        return await gree.search_devices()
```

Comment:
I've looked into the library and version 0.5.0 doesn't implement asyncio correctly. It uses blocking I/O, using a normal socket, inside coroutines. That's not allowed.

We should update to the latest version of the library before moving further with this PR.

Reply:
Done. With the improved async implementation... I think this meets the criteria for Platinum.